### DISCUSSION TITLE
Features from Guilded & Discord Suggestion list

### DISCUSSION THREAD

**Discussion by LunaZE** [stoatchat/discussions#740] [Feature Suggestions]
hello! we migrated from Guilded, and we saw some of the AMAZING features they had. unfortunately, guilded has forced something with accounts that caused a mass exodous. 
Please, consider adding some great features from guilded such as:

1. Category / Groups like Guilded - it allows to make different groups within the server, and makes it incredibly neat
2. Streaming - please add streaming to Revolt
3. Role badge that shows on name of user for Roles
4. Role React bot possibly please!
5. Voice chat that allows you to have many of them in ONE channel such as the listed screenshot
-- Voice that that has a Text chat for those that have Muted mic
6. Please add something like Forums like discord and guilded.
7. Calandar system like Guilded's calandar system

Please take a look at guilded's expansive features and maybe implement some for Revolt, i think it would be an incredibly good thing for revolts growth as well as users will love.

![g2](https://github.com/revoltchat/revolt/assets/85745209/4d6614ac-41db-4132-88cc-43f22b6a6308)
![g1](https://github.com/revoltchat/revolt/assets/85745209/a210fbc3-8131-411f-847f-8e9c9a49c79a)
![g3](https://github.com/revoltchat/revolt/assets/85745209/abba8ad2-a3d6-4a8b-a4ca-be6fe655be1f)
![g4](https://github.com/revoltchat/revolt/assets/85745209/45eb165d-caca-4300-b868-968de0b3e996)

**Comment by randomy1** [comment]
the channel stuff wont be added.

vc will be similar to discord, but in the future you'll be able to stream in vcs, 



forums idk 


role badges are pretty useful, so i could see it happening in like a year


bots are made by users, the revolt team cant just make a bot. 


groups arent being added yet..

**Reply by Rexogamer** [reply]
I'd advise caution with definite statements like these - some of these may become features, and aside from anything I don't see why the team couldn't make a bot...?

**Reply by randomy1** [reply]
yea it might or might not be

**Reply by bitrain574** [reply]
Why won’t channels be added?

**Reply by Rexogamer** [reply]
this is why I'd advise against making definitive statements unless you have official confirmation of some kind 🥲 

to be clear Revolt has channels, though whether it'll gain the various types Guilded has is another matter (and at least imo some of them seem kinda pointless?)

**Reply by bitrain574** [reply]
I think Revolt should have those along with icons of pictures.

**Comment by EmpressAkara** [comment]
I would love to see some or similar features from Guilded make it over to Revolt. I was even thinking of how they could set up the groups. Instead of having the servers on the side bar, the groups could be put there and then the servers could be added in the home section where DMs are. Could just have a "Servers" option there.

There is a role react bot already. It's a third party bot called "Roles".

**Reply by bitrain574** [reply]
That’ll be awesome!

**Reply by bitrain574** [reply]
Yes. Sub servers with pfp pictures are helpful for me.

**Comment by 1NEGROUP** [comment]
Yes I Second This

**Comment by Hiddenpalm** [comment]
My personal thoughts on all of these ideas. I think each one is deserving of its own post.

1. Categories are called Groups in Guilded. Im on the fence about it. I remember it confused some new guildees when joining, because it was different from Discord. On the other hand, I like having the option. I think if it is somehow implemented neater than Guilded, it could work. 

2. Streaming. Would be cool. Streaming from the phone too would be awesome. I recently made a portable retro gaming device with over 22,000 games. Wouldn't mind streaming some of that. If streaming is implemented. Having a special public area on Revolt where anyone can come and see the LIVE public streams, would be a game changer. Watch out Twitch, Revolt is dangerous.

3. Role Badge. Guilded recently did this. And it works with animated emojis. Not a bad idea. I like how you can add emojis to the titles of rooms on Revolt. Guilded can't do that. Allowing for animated emojis to work, would be great. Nerimity allows animated emojis for room titles. Looks pretty cool.

4. Role React Bot. Dont know what that is, so I dont have anything to say on it. I would join a server devoted to bots like RevoltBots and ask there. 

5. Voice Channels that can have multiple channels like Guilded. I'm not against the idea. I have used Guilded for a long time, and I am accustomed to it. Nerimity takes a different approach. Every single text chat room has a voice channel by default. Revolt can one up EVERYONE, by combining these ideas. Every room has the ability to have multiple voice channels! 

For example, one text channel for an MMO guild could be "Open World PVP Channel". That room will automatically have a voice com like Nerimity. But the admin can have Open World PVP Voice as the main voice lobby. And under it, are two more voice channels, Faction A Voice, and Faction B Voice. So there would be a total of 3 voice channels in just that one room, dedicated to open world pvp. 

6. Forums. Yes!!! Absolutely want this. On Guilded, I was able to get every single gaming guide out there and bring it to my guild's server using forums. This way they had every guide, ever, all in one place. This is like the standard now. But what if, every single text chat room, had multiple voice channel capabilities and forums?! That means in the Open World PVP channel that has 3 voice channels, can also have guides and discussions just for Open World PVP of said game using the forums feature. Every text channel by default has voice and forums. 

7. Calendar. YES!!!!!!! Guilded has the best gaming calendar in the whole world wide web. Serious gamers can not live without this feature. Calendars have been a part of computing ever since the very first operating systems. I bet the team at Revolt can nail this one perfectly. 

Of all of them, I really love suggestions 6 and 7. Second would be 5, but a combination of Guilded and Nerimity for voice. I'm cool with the other ideas too.

**Reply by Beanicator** [reply]
Also add connections like Spotify like Discord has.
